bicytok/figures/figure5.py

```diff
@@ -71,31 +71,36 @@ def makeFigure():
             )
             prevOptAffs = optParams[1]
             select = (1 / optParams[0],)
-            
+
             non_marker_columns = ["CellType1", "CellType2", "CellType3", "Cell"]
             marker_columns = CITE_DF.columns[~CITE_DF.columns.isin(non_marker_columns)]
             markerDF = CITE_DF.loc[:, marker_columns]
 
             # Filter to include only columns related to the target receptors
             receptors_of_interest = targets
-            filtered_markerDF = markerDF.loc[:, markerDF.columns.str.contains('|'.join(receptors_of_interest), case=False)]
+            filtered_markerDF = markerDF.loc[
+                :,
+                markerDF.columns.str.contains(
+                    "|".join(receptors_of_interest), case=False
+                ),
+            ]
 
             # Create binary arrays for on-target and off-target cell types
             on_target = (CITE_DF["CellType3"] == targCell).astype(int)
 
             # Define off-target conditions using a dictionary
             off_target_conditions = {
-                0: (CITE_DF["CellType3"] != targCell),     # All non-memory Tregs
-                1: (CITE_DF["CellType2"] != "Treg"),       # All non-Tregs
-                2: (CITE_DF["CellType3"] == "Treg Naive")  # Naive Tregs
+                0: (CITE_DF["CellType3"] != targCell),  # All non-memory Tregs
+                1: (CITE_DF["CellType2"] != "Treg"),  # All non-Tregs
+                2: (CITE_DF["CellType3"] == "Treg Naive"),  # Naive Tregs
             }
 
             # Set off_target based on offTargState
             if offTargState in off_target_conditions:
                 off_target = off_target_conditions[offTargState].astype(int)
             else:
                 raise ValueError("Invalid offTargState value. Must be 0, 1, or 2.")
-            
+
             # Calculate KL divergence and EMD with filtered data and binary arrays
             KLD = KL_divergence_2D(filtered_markerDF, on_target, off_target)
```

Comment:
The outputs of the KL_EMD_2D function are two NxN triangular arrays (one for KL, one for EMD), where N is the number of receptors compared. The diagonals are the 1D distances for each receptor and the off-diagonals are the 2D distances for each receptor pair.

The way this figure is set up, the function is repeated with three receptor pairs. Each time the function outputs a 2x2 triangular array. If I am understanding correctly, in this case, each time you run the function you want a single value for EMD and KL, so you should take index (1,2) of each output array, the 2D value for each pair.